bigchaindb/common/schema/transaction.yaml

```diff
@@ -0,0 +1,257 @@
+        required:
+        - details
+        - uri
+        properties:
+          details:
+            type: object
+            additionalProperties: true
+          uri:
+            type: string
+            pattern: "^cc:([1-9a-f][0-9a-f]{0,3}|0):[1-9a-f][0-9a-f]{0,15}:[a-zA-Z0-9_-]{0,86}:([1-9][0-9]{0,17}|0)$"
+      owners_after:
+        "$ref": "#/definitions/owners_list"
+        description: |
+            List of public keys associated with asset ownership at the time
+            of the transaction.
+      amount:
+        type: integer
+        description: |
+            Integral amount of the asset represented by this condition.
+            In the case of a non divisible asset, this will always be 1.
+  fulfillment:
+    type: "object"
+    description:
+      A fulfillment is an input to a transaction, named as such because it
+      fulfills a condition of a previous transaction. In the case of a
+      ``CREATE`` transaction, a fulfillment may provide no ``input``.
+    additionalProperties: false
+    required:
+    - owners_before
+    - input
+    - fulfillment
+    properties:
+      fid:
+        "$ref": "#/definitions/offset"
+        description: |
+            The offset of the fulfillment within the fulfillents array.
+      owners_before:
+        "$ref": "#/definitions/owners_list"
+        description: |
+            List of public keys of the previous owners of the asset.
+      fulfillment:
+        anyOf:
+        - type: object
+          additionalProperties: false
+          properties:
+            bitmask:
+              type: integer
+            public_key:
+              type: string
+            type:
+              type: string
+            signature:
+              anyOf:
+              - type: string
+              - type: 'null'
+            type_id:
+              type: integer
+          description: |
+              Fulfillment of a condition_, or put a different way, this is a
+              payload that satisfies a condition in order to spend the associated
+              asset.
+        - type: string
+          pattern: "^cf:([1-9a-f][0-9a-f]{0,3}|0):[a-zA-Z0-9_-]*$"
+      input:
+        anyOf:
+        - type: 'object'
+          description: |
+            Reference to a condition of a previous transaction
+          additionalProperties: false
+          properties:
+            cid:
+              "$ref": "#/definitions/offset"
+            txid:
+              "$ref": "#/definitions/sha3_hexdigest"
+        - type: 'null'
+  metadata:
+    anyOf:
+    - type: object
+      description: |
+          User provided transaction metadata. This field may be ``null`` or may
+          contain any valid JSON payload provided by the user.
+      additionalProperties: false
+      required:
+      - id
+      - data
+      properties:
+        data:
+          type: object
+          description: |
+            Freeform object containing user provided asset metadata.
```

Comment:
The metadata data is for this specific transaction. It might not say anything about the asset being transferred. For example, it might say, "There was a thunderstorm in Sydney when Jake Hoopman constructed this transaction."

Maybe just borrow language from the description of the metadata object above.